client/src/language/middlewareDefinition.ts

```diff
@@ -0,0 +1,147 @@
+  } else {
+    return await processDefinitions(tempResult, document, embeddedLanguageDocContent, embeddedLanguageDocInfos)
+  }
+}
+
+const checkIsArrayLocation = (array: Location[] | LocationLink[]): array is Location[] => {
+  return array[0] instanceof Location
+}
+
+const processDefinitions = async <DefinitionType extends Location | LocationLink>(
+  definitions: DefinitionType[],
+  originalTextDocument: TextDocument,
+  embeddedLanguageDocContent: string,
+  embeddedLanguageDocInfos: EmbeddedLanguageDocInfos
+): Promise<DefinitionType[]> => {
+  const result: DefinitionType[] = []
+  await Promise.all(definitions.map(async (definition) => {
+    if (!checkIsDefinitionUriEqual(definition, Uri.parse(embeddedLanguageDocInfos.uri))) {
+      result.push(definition) // only definitions located on the embedded language documents need ajustments
+      return
+    }
+    if (embeddedLanguageDocInfos.language === 'python') {
+      const dDefinition = await getDefinitionOfD(definition)
+      if (dDefinition !== undefined) {
+        result.push(...dDefinition)
+        return
+      }
+      const eDefinition = await getDefinitionOfE(definition)
+      if (eDefinition !== undefined) {
+        result.push(...eDefinition)
+        return
+      }
+    }
+    changeDefinitionUri(definition, originalTextDocument.uri)
+    ajustDefinitionRange(definition, originalTextDocument, embeddedLanguageDocContent, embeddedLanguageDocInfos.characterIndexes)
+    result.push(definition)
+  }))
+  return result
+}
+
+// Map the range of the definitin from the embedded language document to the original document
+const ajustDefinitionRange = (
+  definition: Location | LocationLink,
+  originalTextDocument: TextDocument,
+  embeddedLanguageDocContent: string,
+  characterIndexes: number[]
+): void => {
+  if (definition instanceof Location) {
+    const newRange = getOriginalDocRange(originalTextDocument, embeddedLanguageDocContent, characterIndexes, definition.range)
+    if (newRange !== undefined) {
+      definition.range = newRange
+    }
+  } else {
+    const newTargetRange = getOriginalDocRange(originalTextDocument, embeddedLanguageDocContent, characterIndexes, definition.targetRange)
+    if (newTargetRange !== undefined) {
+      definition.targetRange = newTargetRange
+    }
+    if (definition.targetSelectionRange !== undefined) {
+      const newTargetSelectionRange = getOriginalDocRange(originalTextDocument, embeddedLanguageDocContent, characterIndexes, definition.targetSelectionRange)
+      if (newTargetSelectionRange !== undefined) {
+        definition.targetSelectionRange = newTargetRange
+      }
+    }
+  }
+}
+
+// Redirect a definition to an other definition
+// For example, `d` of `d.getVar('')` is redirected to the definition of `data_smart.DataSmart()`
+const redirectDefinition = async <DefinitionType extends Location | LocationLink>(
```

Comment:
Go to definition works properly in the temporary. Except that yes, as can be expected, d goes to the d assignment and not the definition of the class. So it makes sense to instead point to the definition of DataSmart(). Also with your code, d.getVar points correctly to the getVar function. Perfect!